src/main/java/module/lineup/assistant/LineupAssistant.java

```diff
@@ -53,159 +53,250 @@ public final boolean isPlayerInStartingEleven(int spielerId, Vector<MatchLineupP
 	/**
 	 * Assistant to create automatic lineup
 	 *
-	 * @param lPositions:              list of positions to be filled
+	 * @param allPositions:              list of positions to be filled
 	 * @param lPlayers:                list of available players
 	 * @param sectorsStrengthPriority: priority in sector strength (e.g. MID-FOR-DE)
 	 * @param bForm:                   whether or not to consider the form
 	 * @param idealPosFirst:           whether or not to consider best position first
-	 * @param bInjured:                whether or not to consider injured player
-	 * @param bSuspended:              whether or not to advanced suspended player
+	 * @param ignoreInjured:                whether or not to consider injured player
+	 * @param ignoreSuspended:              whether or not to advanced suspended player
 	 * @param weather:                 Actual weather
 	 */
-	public final void doLineup(List<MatchLineupPosition> lPositions, List<Player> lPlayers,
-							   byte sectorsStrengthPriority, boolean bForm, boolean idealPosFirst, boolean bInjured,
-							   boolean bSuspended, Weather weather) {
+	public final void doLineup(List<MatchLineupPosition> allPositions, List<Player> lPlayers,
+							   byte sectorsStrengthPriority, boolean bForm, boolean idealPosFirst, boolean ignoreInjured,
+							   boolean ignoreSuspended, Weather weather) {
 
-		lPositions = filterPositions(lPositions);
+		var lPositions = allPositions.stream().filter(i -> i.isFieldMatchRoleId() && isSelectedByAssistant(i)).toList();
+		ArrayList<Player> players = new ArrayList<>(lPlayers.stream().filter(i -> (i.getInjuryWeeks() < 1 || ignoreInjured) && (!i.isRedCarded() || ignoreSuspended)).toList());
```

Comment:
```
List<Player> players = lPlayers.stream()  ... .toList();
```
should be enough (i.e. without additional `new ArrayList<>`) – then you just need to change the signature of `optimizeLineup` to take a `List<Player> players`.

Reply:
(or replace `toList()` with `.collect(Collectors.toList())` for a mutable list)

Reply:
mutable list is required - i will try your 2nd suggestion - Thanks!